Update Optimization Module and remove duplicative code (#26565)
idaholab/moose: modules/optimization/doc/content/modules/optimization/examples/forceInv_pointLoads.md

!listing test/tests/optimizationreporter/point_loads/main.i
          block=MultiApps id=multiapps
          caption=Main application `MultiApps` block for point load parameterization shown in [figSetup]

`Transfers` makes up the next section of the main input file.  TAO interacts
with the various sub-apps using the reporter system and the

> [MultiAppReporterTransfer.md].  The first three `Transfers` in [transfers]

Reviewer: ```suggestion
[MultiAppReporterTransfer.md].  The first two `Transfers` in [transfers]
```

communicate data with the forward sub-app.  These `Transfers` are used by TAO to

> compute the objective. The first transfer `[toForward_measument]` communicates
> the measurement locations to the forward app object [OptimizationData.md].

Reviewer: ```suggestion
compute the objective. The first transfer `[toForward_measument]` communicates
the measurement locations to the forward app object [OptimizationData.md].  This transfer can be avoided by placing the [OptimizationData.md] block on the forward solve input file instead of the main optimization input file.
```

> The next set of `Transfers` in [transfers] communicate reporter values between the main-app and adjoint sub-app to compute the gradient of the objective function with respect to the controllable parameter.  The fourth transfer named `[toAdjoint]` sends the misfit data at each measurement point to an [OptimizationData.md] reporter on the adjoint-app which is then consumed by the [ReporterPointSource.md] dirackernel to apply the misfit source loading as described by the top equation in [!eqref](theory/InvOptTheory.md#eqn:adjoint_problem).  The fifth transfer named `[fromAdjoint]` retrieves the gradient from the adjoint-app for TAO.

Reviewer: ```suggestion
The next set of `Transfers` in [transfers] communicate reporter values between the main-app and adjoint sub-app to compute the gradient of the objective function with respect to the controllable parameter.  The third transfer named `[toAdjoint]` sends the misfit data at each measurement point to an [OptimizationData.md] reporter on the adjoint-app which is then consumed by the [ReporterPointSource.md] dirackernel to apply the misfit source loading as described by the top equation in [!eqref](theory/InvOptTheory.md#eqn:adjoint_problem).  The fourth transfer named `[fromAdjoint]` retrieves the gradient from the adjoint-app for TAO.  The sibling transfer system could have been used to transfer the misfit data directly from the forward solve to the adjoint solve.
```

The final set of `Transfers` in [transfers] communicate reporter values between
the main-app and the homogeneous forward sub-app to compute a matrix free
Hessian.  The homogeneous forward transfers, `[toHomoForward_measument]`
`[toHomoForward]` and `[fromHomoForward]` transfer the same reporter data to the
same objects types used by the forward-app.  For force inversion, the
homogeneous forward app solves the same kernels as the forward-app except the
boundary conditions are homogenized, i.e. set to zero.  The parameter being
controlled in [OptimizationReporter](syntax/OptimizationReporter/index.md) by
the [Optimize.md] executioner for the homogeneous forward problem is the
variation in the parameter which is determined by TAO in order to build up the
required matrix free Hessian. In the `Transfer` from the homogenous forward
sub-app [homoForward], the `simulation_values` are transferred into the
`misfit_values` in the main application. This will be problem specific to the

> tpy eof optimization performed.

Reviewer: ```suggestion
type of optimization performed.  It has only been implemented for force inversion problems.
```
